src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -677,5 +723,49 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 throw new NotSupportedException();
             }
         }
+
+        private class DuplexPipeConverter : JsonConverter<IDuplexPipe>
+        {
+            private readonly JsonMessageFormatter jsonMessageFormatter;
+
+            public DuplexPipeConverter(JsonMessageFormatter jsonMessageFormatter)
+            {
+                this.jsonMessageFormatter = jsonMessageFormatter ?? throw new ArgumentNullException(nameof(jsonMessageFormatter));
+            }
+
+            public override IDuplexPipe ReadJson(JsonReader reader, Type objectType, IDuplexPipe existingValue, bool hasExistingValue, JsonSerializer serializer)
+            {
+                int? tokenId = JToken.Load(reader).Value<int?>();
```

Comment:
Would it be better to serialize as an object, e.g. `{"id": id}` for future extensibility?

Reply:
I considered that, but I couldn't think of any extensibility we would need. This depends on MultiplexingStream, which only needs an integer to accept a channel. And if we ever switched away from MultiplexingStream, that would be a breaking change on both sides so we would have to define a new protocol for whatever would replace it anyway.